[PATCH 29/37] board: secure boot: common: Add config for secure boot

[Description]
Add config for secure boot
	FIT_KEY: path to fit public key
	Select algo: sha256,rsa2048 sha256,rsa3072 sha384,rsa3072

Change-Id: I6b71dd5795c4bb63f01d1c17c9715e4cad3f3220
---
 Makefile               | 18 +++++++-----------
 board/mediatek/Kconfig | 26 ++++++++++++++++++++++++++
 2 files changed, 33 insertions(+), 11 deletions(-)
--- a/Makefile
+++ b/Makefile
@@ -1165,16 +1165,12 @@ endif
 	$(call cmd,ofcheck,$(KCONFIG_CONFIG))
 
 
-ifneq ($(FIT_KEY),)
-FIT_KEY_DIR := $(patsubst %/,%,$(dir $(FIT_KEY)))
-FIT_KEY_NAME := $(basename $(notdir $(FIT_KEY)))
-
-ifneq ($(KEY_ALG),)
-FIT_KEY_ALG := $(KEY_ALG)
-else
-FIT_KEY_ALG := sha256,rsa2048
+ifeq ($(CONFIG_MTK_SECURE_BOOT),y)
+ifeq ($(CONFIG_FIT_KEY),)
+	$(error "Must set FIT_KEY when enable secure boot")
 endif
-
+FIT_KEY_DIR := $(shell dirname $(CONFIG_FIT_KEY))
+FIT_KEY_NAME := $(shell basename $(CONFIG_FIT_KEY) | cut -d '.' -f 1)
 endif
 
 PHONY += dtbs dtbs_check
@@ -1182,8 +1178,8 @@ dtbs: dts/dt.dtb
 	@:
 dts/dt.dtb: dtbs_prepare u-boot
 	$(Q)$(MAKE) $(build)=dts dtbs
-ifneq ($(FIT_KEY),)
-	$(objtree)/tools/fdt_add_pubkey -a $(FIT_KEY_ALG) -k $(FIT_KEY_DIR) -n $(FIT_KEY_NAME) -r conf $@
+ifneq ($(CONFIG_MTK_SECURE_BOOT),)
+	$(objtree)/tools/fdt_add_pubkey -a $(CONFIG_KEY_ALG) -k $(FIT_KEY_DIR) -n $(FIT_KEY_NAME) -r conf $@
 endif
 
 ifeq ($(CONFIG_MTK_FW_ENCRYPT),y)
--- a/board/mediatek/Kconfig
+++ b/board/mediatek/Kconfig
@@ -269,6 +269,32 @@ config CHAIN_OF_TRUST
 	bool
 	default n
 
+config FIT_KEY
+	string "FIT PUBLIC KEY"
+	default "../keys/fit_key.crt"
+	help
+	  Set FIT_KEY for secure boot.
+
+choice
+	prompt "Secure Boot Algorithm"
+	default SBC_ALGO_RSA2048_SHA256
+
+	config SBC_ALGO_RSA2048_SHA256
+		bool "rsa2048+sha256"
+
+	config SBC_ALGO_RSA3072_SHA256
+		bool "rsa3072+sha256"
+
+	config SBC_ALGO_RSA3072_SHA384
+		bool "rsa3072+sha384"
+endchoice
+
+config KEY_ALG
+	string
+	default "sha256,rsa2048" if SBC_ALGO_RSA2048_SHA256
+	default "sha256,rsa3072" if SBC_ALGO_RSA3072_SHA256
+	default "sha384,rsa3072" if SBC_ALGO_RSA3072_SHA384
+
 config MTK_ANTI_ROLLBACK
 	bool "Enable MediaTek anti-rollback framework"
 	default n
